internal/venice-common/src/main/java/com/linkedin/venice/pubsub/adapter/kafka/consumer/ApacheKafkaConsumerAdapter.java

```diff
@@ -103,7 +103,7 @@ public void subscribe(PubSubTopicPartition pubSubTopicPartition, long lastReadOf
     }
 
     // Check if the topic-partition exists
-    if (!isValidTopicPartition(pubSubTopicPartition)) {
+    if (config.shouldCheckTopicExistenceBeforeConsuming() && !isValidTopicPartition(pubSubTopicPartition)) {
```

Comment:
I thought we cannot skip this check?

Reply:
Ideally we should not. Added this check so that if we need to revert back to Kafka clients and want to have old behavior due to mitigate unknown issues

Reply:
I see.